src/components/implementation/no_interface/new_boot/ppos_booter.c

```diff
@@ -0,0 +1,323 @@
+/*
+ *TODO:
+ *	Test Scheduling Array for dependencies 
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <cos_component.h>
+#include <cobj_format.h>
+#include <cos_kernel_api.h>
+#include <boot_deps.h>
+
+/*Assembly function for sinv from new component*/
+extern void *__inv_test_entry(int a, int b, int c);
+struct cos_compinfo boot_info;	
+
+struct cobj_header *hs[MAX_NUM_SPDS+1];
+
+struct deps {
+	short int client, server;
+};
+struct deps *deps;
+int ndeps;
+
+static void
+cos_llprint(char *s, int len)
+{ call_cap(PRINT_CAP_TEMP, (int)s, len, 0, 0); }
+
+int
+prints(char *s)
+{
+		int len = strlen(s);
+		cos_llprint(s, len);
+		return len;
+}
+
+int __attribute__((format(printf,1,2)))
+printc(char *fmt, ...)
+{
+	  char s[128];
+  	  va_list arg_ptr;
+  	  int ret, len = 128;
+
+  	  va_start(arg_ptr, fmt);
+  	  ret = vsnprintf(s, len, fmt, arg_ptr);
+  	  va_end(arg_ptr);
+	  cos_llprint(s, ret);
+
+	  return ret;
+}
+
+/*Component init info*/
+#define INIT_STR_SZ 52
+struct component_init_str {
+        unsigned int spdid, schedid;
+        int startup;
+        char init_str[INIT_STR_SZ];
+}__attribute__((packed));
+
+struct component_init_str *init_args;
+
+unsigned int *boot_sched;
+
+static void
+boot_find_cobjs(struct cobj_header *h, int n)
+{
+	int i;
+	vaddr_t start, end;
+
+	start = (vaddr_t)h;
+	hs[0] = h;
+
+	for (i = 1 ; i < n ; i++) {
+		int j = 0, size = 0, tot = 0;
+
+		size = h->size;
+		for (j = 0 ; j < (int)h->nsect ; j++) {
+			tot += cobj_sect_size(h, j);
+		}
+		printc("cobj %s:%d found at %p:%x, size %x -> %x\n",
+		       h->name, h->id, hs[i-1], size, tot, cobj_sect_get(hs[i-1], 0)->vaddr);
+
+		end   = start + round_up_to_cacheline(size);
+		hs[i] = h = (struct cobj_header*)end;
+		start = end;
+	}
+	
+	hs[n] = NULL;
+	printc("cobj %s:%d found at %p -> %x\n",
+	       hs[n-1]->name, hs[n-1]->id, hs[n-1], cobj_sect_get(hs[n-1], 0)->vaddr);
+}
+
+static vaddr_t
+boot_spd_end(struct cobj_header *h)
+{
+	struct cobj_sect *sect;
+	int max_sect;
+
+	max_sect = h->nsect-1;
+	sect     = cobj_sect_get(h, max_sect);
+
+	return sect->vaddr + round_up_to_page(sect->bytes);
+}
+
+static int
+boot_spd_symbs(struct cobj_header *h, spdid_t spdid, vaddr_t *comp_info)
+{
+//	printc("boot_spd_symbs: %d \n", h->nsymb);
```

Comment:
Note that you didn't check off any of the checkboxes on the PR template for code craftsmanship.  In your revisions, please do take care of all of that, and correspondingly check of the boxes.